// finnhub/dist/model/COVID19.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Covid19 model module.
 * @module model/Covid19
 * @version 1.1.1
 */
var Covid19 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Covid19</code>.
   * @alias module:model/Covid19
   */
  function Covid19() {
    _classCallCheck(this, Covid19);

    Covid19.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Covid19, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Covid19</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Covid19} obj Optional instance to populate.
     * @return {module:model/Covid19} The populated <code>Covid19</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Covid19();

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('case')) {
          obj['case'] = _ApiClient["default"].convertToType(data['case'], 'Number');
        }

        if (data.hasOwnProperty('death')) {
          obj['death'] = _ApiClient["default"].convertToType(data['death'], 'Number');
        }

        if (data.hasOwnProperty('updated')) {
          obj['updated'] = _ApiClient["default"].convertToType(data['updated'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return Covid19;
}();
/**
 * State.
 * @member {String} state
 */


Covid19.prototype['state'] = undefined;
/**
 * Number of confirmed cases.
 * @member {Number} case
 */

Covid19.prototype['case'] = undefined;
/**
 * Number of confirmed deaths.
 * @member {Number} death
 */

Covid19.prototype['death'] = undefined;
/**
 * Updated time.
 * @member {Date} updated
 */

Covid19.prototype['updated'] = undefined;
var _default = Covid19;
exports["default"] = _default;
