// finnhub/dist/model/Splits.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Splits model module.
 * @module model/Splits
 * @version 1.1.0
 */
var Splits = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Splits</code>.
   * @alias module:model/Splits
   */
  function Splits() {
    _classCallCheck(this, Splits);

    Splits.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Splits, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Splits</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Splits} obj Optional instance to populate.
     * @return {module:model/Splits} The populated <code>Splits</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Splits();

        if (data.hasOwnProperty('symbol')) {
          obj['symbol'] = _ApiClient["default"].convertToType(data['symbol'], 'String');
        }

        if (data.hasOwnProperty('Date')) {
          obj['Date'] = _ApiClient["default"].convertToType(data['Date'], 'Date');
        }

        if (data.hasOwnProperty('fromFactor')) {
          obj['fromFactor'] = _ApiClient["default"].convertToType(data['fromFactor'], 'Number');
        }

        if (data.hasOwnProperty('toFactor')) {
          obj['toFactor'] = _ApiClient["default"].convertToType(data['toFactor'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Splits;
}();
/**
 * Symbol.
 * @member {String} symbol
 */


Splits.prototype['symbol'] = undefined;
/**
 * Split date.
 * @member {Date} Date
 */

Splits.prototype['Date'] = undefined;
/**
 * From factor.
 * @member {Number} fromFactor
 */

Splits.prototype['fromFactor'] = undefined;
/**
 * To factor.
 * @member {Number} toFactor
 */

Splits.prototype['toFactor'] = undefined;
var _default = Splits;
exports["default"] = _default;
