// finnhub/dist/model/Trade.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Trade model module.
 * @module model/Trade
 * @version 1.1.4
 */
var Trade = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Trade</code>.
   * @alias module:model/Trade
   */
  function Trade() {
    _classCallCheck(this, Trade);

    Trade.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Trade, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Trade</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Trade} obj Optional instance to populate.
     * @return {module:model/Trade} The populated <code>Trade</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Trade();

        if (data.hasOwnProperty('s')) {
          obj['s'] = _ApiClient["default"].convertToType(data['s'], 'String');
        }

        if (data.hasOwnProperty('p')) {
          obj['p'] = _ApiClient["default"].convertToType(data['p'], 'Number');
        }

        if (data.hasOwnProperty('t')) {
          obj['t'] = _ApiClient["default"].convertToType(data['t'], 'Number');
        }

        if (data.hasOwnProperty('v')) {
          obj['v'] = _ApiClient["default"].convertToType(data['v'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Trade;
}();
/**
 * Symbol.
 * @member {String} s
 */


Trade.prototype['s'] = undefined;
/**
 * Last price.
 * @member {Number} p
 */

Trade.prototype['p'] = undefined;
/**
 * UNIX milliseconds timestamp.
 * @member {Number} t
 */

Trade.prototype['t'] = undefined;
/**
 * Volume.
 * @member {Number} v
 */

Trade.prototype['v'] = undefined;
var _default = Trade;
exports["default"] = _default;
